File: sort/sort_01.c
#include <stdio.h>

typedef struct {
  const char *name;
  int x, y;
} City;

City cities[] = {
    {"Clean", 1336, 536}, {"Prosy", 977, 860},  {"Rabbi", 6, 758},    {"Addle", 222, 261},
    {"Smell", 1494, 836}, {"Quite", 905, 345},  {"Lives", 72, 714},   {"Cross", 23, 680},
    {"Synth", 1529, 785}, {"Tweak", 1046, 426}, {"Medic", 1485, 514}, {"Glade", 660, 476},
    {"Breve", 1586, 448}, {"Hotel", 1269, 576}, {"Toing", 398, 561},  {"Scorn", 617, 373},
    {"Tweet", 1253, 403}, {"Zilch", 1289, 29},  {"React", 296, 659},  {"Fiche", 787, 278},
};

void City_print(City *c)
{
  printf("%s(%d,%d)", c->name, c->x, c->y);
}
void City_printAll(City *p, int count)
{
  for (int i = 0; i < count; i++, p++) {
    City_print(p);
    putchar(' ');
  }
  putchar('\n');
}

int main(void) 
{
  printf("--- Original Data ---\n");
  City_printAll(cities, sizeof(cities) / sizeof(cities[0]));

  // sort here by name
  printf("--- By Name ---\n");
  City_printAll(cities, sizeof(cities) / sizeof(cities[0]));

  // sort here by y coordinate
  printf("--- By Y Coordinate ---\n");
  City_printAll(cities, sizeof(cities) / sizeof(cities[0]));

  return 0;
}

File: sort/sort_01.js
// https://replit.com/languages/nodejs

class City {
  constructor(name, x, y) {
    this.name = name;
    this.x = x;
    this.y = y;
  }
}

let cities = [
  new City("Clean", 1336, 536),  new City("Prosy", 977, 860),
  new City("Rabbi", 6, 758),     new City("Addle", 222, 261),
  new City("Smell", 1494, 836),  new City("Quite", 905, 345),
  new City("Lives", 72, 714),    new City("Cross", 23, 680),
  new City("Synth", 1529, 785),  new City("Tweak", 1046, 426),
  new City("Medic", 1485, 514),  new City("Glade", 660, 476),
  new City("Breve", 1586, 448),  new City("Hotel", 1269, 576),
  new City("Toing", 398, 561),   new City("Scorn", 617, 373),
  new City("Tweet", 1253, 403),  new City("Zilch", 1289, 29),
  new City("React", 296, 659),   new City("Fiche", 787, 278),
]
console.log("--- Original Data ---")
console.log(cities)

// sort here by name
console.log("--- By Name ---")
console.log(cities)

// sort here by x
console.log("--- By X Coordinate ---")
console.log(cities)
